tsconfig.json

```diff
@@ -3,7 +3,14 @@
   "compilerOptions": {
     "target": "es5",
     "module": "commonjs",
+    "moduleResolution": "node",
     "outDir": "./dist",
-    "strict": true
+    "strict": true,
+    "declaration": true,
```

Comment:
Did we need all these changes?

Reply:
@royliu2's guidance was to include these

Reply:
Kk. I usually avoid settings that are not needed, and we're unsure if we even need 😄

Reply:
i would be in favor of keeping whichever configs here make sense for us in this package. so `declaration` to have some generated type defs, unsure about the others below this line but they seem like sane config defaults to me.